src/xdsl/parser.py

```diff
@@ -377,6 +377,26 @@ def parse_optional_string(self,
 
     T = TypeVar('T')
 
+    def parse_nested_list(self,
```

Comment:
This just parses this structure if using the mlir parser, right? Could we also have this for the xDSL one? Maybe ask @math-fehr how this should look (probably just the same IMO).

Reply:
Yeah, it would be nice to have them be in sync, I'll do this now